cpp/src/parquet/properties.cc

```diff
@@ -28,16 +28,20 @@
 namespace parquet {
 
 std::shared_ptr<ArrowInputStream> ReaderProperties::GetStream(
-    std::shared_ptr<ArrowInputFile> source, int64_t start, int64_t num_bytes) {
-  if (buffered_stream_enabled_) {
+    std::shared_ptr<ArrowInputFile> source, int64_t start, int64_t num_bytes,
+    std::optional<int64_t> custom_buffer_size) {
```

Comment:
Emmm why do we need a `custom_buffer_size` even when we already have a `buffer_size` internal? This makes building a stream has "default buffer size", "buffer size in ReaderProperties", "buffer size in GetStream"...